docker/pypi/wmagent/run.sh

```diff
@@ -0,0 +1,683 @@
+        echo "$FUNCNAME: Certifficate end date: $certEndDate"
+        [[ -z $certEndDate ]] && { echo "ERROR: Failed to determine certificate end date!"; return $(false) ;}
+
+        certEndDate=$(date --date="$certEndDate" +%s)
+        [[ $certEndDate -le $now ]] && { echo "ERROR: Expired certificate at $WMA_CERTS_DIR/servicecert.pem!"; return $(false) ;}
+        [[ $(($certEndDate -$now)) -le $certMinLifetimeSec ]] && { echo "WARNING: The service certificate lifetime is less than certMinLifetimeHours: $certMinLifetimeHours! Please update it ASAP!" ;}
+
+        # Renew myproxy if needed:
+        echo "$FUNCNAME: Checking myproxy lifetime:"
+        local myproxyEndDate=$(openssl x509 -in $WMA_CERTS_DIR/myproxy.pem -noout -enddate)
+        myproxyEndDate=${myproxyEndDate##*=}
+        echo "$FUNCNAME: myproxy end date: $myproxyEndDate"
+        [[ -n $myproxyEndDate ]] || ($WMA_DEPLOY_DIR/deploy/renew_proxy.sh) || { echo "ERROR: Failed to renew invalid myproxy"; return $(false) ;}
+
+        myproxyEndDate=$(date --date="$myproxyEndDate" +%s)
+        [[ $myproxyEndDate -gt $now ]] || ($WMA_DEPLOY_DIR/deploy/renew_proxy.sh) || { echo "ERROR: Failed to renew expired myproxy"; return $(false) ;}
+
+        # Stay safe and always change the service {cert,key} and myproxy mode here:
+        sudo chmod 600 $WMA_CERTS_DIR/*
+        echo "$FUNCNAME: OK"
+    else
+        echo "ERROR: We found no service certificate installed at $WMA_CERTS_DIR!"
+        echo "ERROR: Please install proper cert and key files before restarting the WMAgent container!"
+        return $(false)
+    fi
+
+    # Update flavor/type/domain global variables if needed
+    # (grep -E "^[[:blank:]]*ORACLE_" $WMA_ADMIN_DIR/WMAgent.secrets > /dev/null) && CURR_FLAVOR=oracle || CURR_FLAVOR=mysql
+
+    echo "Done: $stepMsg"
+    echo "-------------------------------------------------------"
+}
+
+_check_oracle()
+{
+    # Auxiliary finction to check if the oracle database configured for the current agent is empty
+    # NOTE: Oracle is centraly provided - we require an empty database for every account/agent
+    #       otherwise we cannot guarantie this is the only agent to connect to the so configured database
+    echo "$FUNCNAME: Checking whether the oracle database is clean and not used by other agents ..."
+    $manage db-prompt<<"    EOF">/tmp/db_check_output
+    SELECT COUNT(*) from USER_TABLES;
+    EOF
+    tables=`cat /tmp/db_check_output | grep -A1 '\-\-\-\-' | tail -n 1`
+    rm -rf /tmp/db_check_output
+    if [ "$tables" -gt 0 ]; then
+        echo "ERROR: Non empty database found: $tables tables."; return $(false)
+    else
+        echo "OK"; return $(true)
+    fi
+}
+
+_check_mysql()
+{
+    # Auxiliary finction to check if the the MariaDB database for the current agent is properly set
+    # TODO: To be implemented in the issue related to the MariaDB setup fro wmagent
+    echo "$FUNCNAME: Checking whether the mysql schema has been installed"
+    true
```

Comment:
Is it a work in progress?

Reply:
This is a placeholder here.  It should be worked out in this GH issue:  https://github.com/dmwm/WMCore/issues/11313